protected/controllers/AdminDatasetController.php

```diff
@@ -243,11 +243,10 @@ public function actionUpdate($id)
                 $this->renderNotificationsAccordingToStatus($uploadStatus, $previousUploadStatus, $model);
             }
 
-            // semantic kewyords update, using remove all and re-create approach
-            if ($postKeywords = Yii::$app->request->post('keywords')) {
-                $attribute_service = Yii::app()->attributeService;
-                $attribute_service->replaceKeywordsForDatasetIdWithString($id, $postKeywords);
-            }
+            // semantic keywords update, using remove all and re-create approach
+            $postKeywords = Yii::$app->request->post('keywords', '');
+            $attribute_service = Yii::app()->attributeService;
+            $attribute_service->replaceKeywordsForDatasetIdWithString($id, $postKeywords);
```

Comment:
**suggestion(non-blocking):** Might be beneficial to add validation logic to handle cases where empty strings should not be processed further, for example:
```
$postKeywords = Yii::$app->request->post('keywords', '');
if (!empty($postKeywords)) {
    $attribute_service = Yii::app()->attributeService;
    $attribute_service->replaceKeywordsForDatasetIdWithString($id, $postKeywords);
}
```
This makes sure that empty keyword will not trigger any action in `attributeService` and `replaceKeywordsForDatasetIdWithString()`.

Reply:
if I do this, won't it disable deleting keywords? I mean, this undoes the changes. I think the idea is to make it possible for the user to remove all keywords

With these lines, deleting the keywords and saving still shows the keywords that were attempted to delete

Please clarify

Reply:
It seems that adding this condition only works for the case if there is no keyword exist initially, so please ignore this suggestion.